webrtc.html

```diff
@@ -9712,7 +9712,41 @@ <h2>Attributes</h2>
             <dt><dfn data-idl><code>canInsertDTMF</code></dfn> of type <span class=
             "idlAttrType"><a>boolean</a></span>, readonly</dt>
             <dd>
-              <p>Whether the <a>RTCDTMFSender</a> is capable of sending DTMF.</p>
+              <p>Whether the <a>RTCDTMFSender</a> <var>dtmfSender</var> is capable of
+              sending DTMF. On getting, the attribute MUST return the value of
+              <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot. To set
+              <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot, run the following steps
+              when <code><var>pc</var>.onsignalingstatechange</code> or
+              <code><var>pc</var>.connectionstatechange</code> is fired:
```

Comment:
Perhaps we want to phrase this a bit differently. Rather than saying it should be done "when onsignalingstatechange fires" it seems to me it should be expressed in terms of "update the connection state" (dfn) and "signaling-state" (dfn) changes.

Reply:
I worry this reads like a shim, and I hope no-one implements it this way. It's also not clear what thread this is to happen on. Internal slots lets us move setting to the respective algorithms that will end up needing to set it. If the number of places needed to references this are too numerous, we could perhaps break out a sub-algorithm like we did for [update the negotiation-needed flag](https://w3c.github.io/webrtc-pc/#dfn-update-the-negotiation-needed-flag), but we still need to be specific about when it is called, in the implementation itself, and not piggy-back on other JS observables IMHO.

Seeing the below, I also worry there's too much state baked into this attribute...